### DISCUSSION TITLE
How to burn iPXE to the ROM of Mellanox ConnectX-3 card ?

### DISCUSSION THREAD

**Discussion by janisbit** [ipxe/ipxe#1065] [Q&A]
Hello,

does anyone know if it is possible to burn an ipxe driver into the expansion ROM of Mellanox Connect-X3 card (MT27500 Family ) ?
The driver I would need is 15b31003  in EFI mode.
If is it possible to do it, does anyone know how to build it (which make commands and additional steps) in order to be in in the right format to make it burnable to the card's ROM?
If I understood correctly it need to be signed?  Any pointers to how to do it?

Thanks in advance for your help.
best regards,
Janis

**Comment by dfh0522** [comment]
I just burned my HP 544 Connect-X3 to ipxe last month in pcbios mode, it's working without problem.

make EMBED=/home/dfh0522/boot.ipxe EXTRA_CFLAGS="-fcommon"  bin/15b31003.rom

Also I tried to follow [https://ipxe.org/appnote/buildtargets](url), it's successful to make, but didn't try to burn it.

make EMBED=/home/dfh0522/boot.ipxe EXTRA_CFLAGS="-fcommon"  bin-x86_64-efi/15b31003.efi

**Comment by janisbit** [comment]
Hello dfh0522,
thank you for your feedback.

Could you please write the exact command with all parameters (flint, mlxburn etc)  needed in order to correctly burn the ROM image to the card?

thanks,
Janis

**Reply by dfh0522** [reply]
Sure, just git clone master branch to build bin/15b31003.rom, then burn it:

flint -d mt4099_pci_cr0 --allow_rom_change brom 15b31003.rom
